cloudsync/sync/manager.py

```diff
@@ -1339,6 +1339,10 @@ def embrace_change(self, sync, changed, synced):  # pylint: disable=too-many-ret
             return self.delete_synced(sync, changed, synced)
 
         if sync[changed].exists == MISSING:
+            if sync[synced].exists == EXISTS and not sync[synced].changed:
+                sync[synced].changed = time.time()
+                log.warning("%s missing, other side exists. Marking other side changed.")
+                # do we want to punt here, or fall through to marking finished?
```

Comment:
it doesn't seem necessary to punt, since the side you're marking finished is the side that has a "missing" file.

Reply:
Mike added the PUNT here:

if sync.priority <= 2:
    log.warning("%s missing, other side exists. punting: %s", sync[changed].path, sync)
    return PUNT